frappe/contacts/doctype/address/address.py

```diff
@@ -145,30 +145,10 @@ def get_list_context(context=None):
 def get_address_list(doctype, txt, filters, limit_start, limit_page_length = 20, order_by = None):
 	from frappe.www.list import get_list
 	user = frappe.session.user
-	ignore_permissions = False
-	if is_website_user():
-		if not filters: filters = []
-		add_name = []
-		contact = frappe.db.sql("""
-			select
-				address.name
-			from
-				`tabDynamic Link` as link
-			join
-				`tabAddress` as address on link.parent = address.name
-			where
-				link.parenttype = 'Address' and
-				link_name in(
-				   select
-					   link.link_name from `tabContact` as contact
-				   join
-					   `tabDynamic Link` as link on contact.name = link.parent
-				   where
-					   contact.user = %s)""",(user))
-		for c in contact:
-			add_name.append(c[0])
-		filters.append(("Address", "name", "in", add_name))
-		ignore_permissions = True
+	ignore_permissions = True
+
+	if not filters: filters = []
+	filters.append(("Address", "owner", "=", user))
```

Comment:
What about system managers and administrators?

Is this function used for list_view as well?

Reply:
This function is used only to render the list view in web forms. In those cases, System Managers and Administrator accounts would be displayed all the addresses in the system. And in case a desk user didn't have access to the `Address` DocType, they would not be able to view `/address` however a Website User could still view them. 

With this current change, users will only use (at the standard address web form) the addresses they've added

Reply:
Just call the get_list anyway and let role permission manager handle the rest

Reply:
if i do that, nobody except System Managers will be able to see their own Addresses